app/src/main/java/com/github/se/travelpouch/model/ListTravelViewModel.kt

```diff
@@ -83,5 +104,34 @@ open class ListTravelViewModel(private val repository: TravelRepository) : ViewM
    */
   fun selectTravel(travel: TravelContainer) {
     selectedTravel_.value = travel
+    participants_.value = emptyMap()
+  }
+
+  fun fetchAllParticipantsInfo() {
```

Comment:
Yes this is a pertinent observation. This might be as you say unduly burdensome. My initial idea was that if one is editing a travel, they will switch back and forth between the edit travel screen and the participant list screen, checking for roles.

This might not happen often enough to warrant this extra logic. Since we're using blaze billing mode, it might be useful to control the amount of queries. But more importantly, it reduces latency for the user. Not having to wait 200ms on IO for all participants to appear if we have a lot of them gives the user a much smoother experience.